app/components/meetings/sessions/session-list.vue

```diff
@@ -38,42 +38,83 @@
 
 <script>
 
-import   Api, {mapObjectId}from '../api.js'
-import { dateTimeFilter  } from '../filters.js'
+import   Api           , { mapObjectId }from '../api.js'
+import { dateTimeFilter }               from '../filters.js'
+import   moment                         from 'moment'
 
 export default {
   name       : 'SessionsList',
   props      : { 
                   route:       { type: Object, required: false },
                   tokenReader: { type: Function, required: false }
                 },
+  methods    : { getUrl, meetingQuery,conferenceQuery, sessionNumber, isInProgress, now },
+  computed   : { now },
   filters    : { dateTimeFilter },
-  created, 
-  data,
-  methods: { getUrl} 
+  created, data
 }
 
 function data(){
   return { 
     sessions: [],
-    meetings: []
+    meetings: [], 
+    api: undefined
   }
 }
 
-function getUrl(id) {
-  return `/meetings/${encodeURIComponent(this.route.params.meeting)}/sessions/${encodeURIComponent(id)}`
+function getUrl({ _id, meetings }) {
+  const { symbol } = meetings[0]
+
+  return `/meetings/${encodeURIComponent(symbol)}/sessions/${encodeURIComponent(_id)}`
 }
 
 async function created(){
   this.api = new Api(this.tokenReader);
 
-  let q = { normalizedSymbol: this.route.params.meeting.toUpperCase() };
-  const meetings = await this.api.queryMeetings({q, f: { _id:1, normalizedSymbol:1 }});
+  const isMeeting = await this.meetingQuery();
+  const q         = isMeeting? isMeeting: await this.conferenceQuery();
+
+  this.sessions  = await this.api.querySessions({ q, s: { date:-1 } });
+}
+
+async function meetingQuery(){
+  const normalizedSymbol = this.route?.params?.meeting? this.route?.params?.meeting.toUpperCase() : ''
```

Comment:
this syntax is so confusing

Reply:
I agree why not just simply:
```suggestion
  const normalizedSymbol = this.route?.params?.meeting?.toUpperCase() || '';
```

Reply:
I feel the '?.'  is not being parsed as it should.  I think my original was simply this.route?.params?.meeting?.toUpperCase() alone which should suffice.  However, error on ?.toUpperCase() would still occur.

Reply:
only if meeting if is set and not a string (not have toUpperCase as function)

```javascript
> var a = { b: 1 }
> a?.b
  1
> a?.b?.c?.toUpperCase()
  undefined
a?.b?.toUpperCase()
  Uncaught TypeError: a?.b?.toUpperCase is not a function
```